### DISCUSSION TITLE
ASP.NET Core integration tests with NUnit and Moq

### DISCUSSION THREAD

**Discussion by giscus** [CamiloTerevinto/Blog#4] [Announcements]
# ASP.NET Core integration tests with NUnit and Moq

https://www.camiloterevinto.com/post/asp-net-core-integration-tests-with-nunit-and-moq

**Comment by devodo** [comment]
Hello and thank you for demonstrating this useful approach. I also find it very valuable to be able to mock services like this. Moq can be a good option but it did not meet all my needs and I wanted a cleaner solution so I created a library that might be of interest: https://github.com/devodo/DivertR
It is custom built to integrate with the dependency injection container and WebApplicationFactory.
